f5lbaasdriver/test/tempest/tests/api/test_l7policy.py

```diff
@@ -0,0 +1,143 @@
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from neutron.plugins.common import constants as plugin_const
+from neutron_lbaas.services.loadbalancer import constants as lb_const
+from tempest import config
+from tempest.lib.common.utils import data_utils
+from tempest import test
+
+from f5lbaasdriver.test.tempest.tests.api import base
+
+
+CONF = config.CONF
+
+
+class L7PolicyTestJSON(base.BaseTestCase):
+    """L7 Policy tempest tests.
+
+    Tests the following operations in the Neutron-LBaaS API using the
+    REST client with default credentials:
+
+    1) Creating a L7 policy with REJECT action.
+    2) Creating a L7 policy with a REDIRECT_URL action.
+    3) Creating a L7 policy with a REDIRECT_POOL action.
+    """
+
+    @classmethod
+    def resource_setup(cls):
+        super(L7PolicyTestJSON, cls).resource_setup()
+        if not test.is_extension_enabled('lbaasv2', 'network'):
+            msg = "lbaas extension not enabled."
+            raise cls.skipException(msg)
+        network_name = data_utils.rand_name('network')
+        cls.network = cls.create_network(network_name)
+        cls.subnet = cls.create_subnet(cls.network)
+        cls.create_lb_kwargs = {'tenant_id': cls.subnet['tenant_id'],
+                                'vip_subnet_id': cls.subnet['id']}
+        cls.load_balancer = \
+            cls._create_active_load_balancer(**cls.create_lb_kwargs)
+        cls.load_balancer_id = cls.load_balancer['id']
+
+        # Create listener for tests
+        cls.create_listener_kwargs = {'loadbalancer_id': cls.load_balancer_id,
+                                      'protocol': "HTTP",
+                                      'protocol_port': "80"}
+        cls.listener = (
+            cls._create_listener(**cls.create_listener_kwargs))
+        cls.listener_id = cls.listener['id']
+
+        # Create pool for tests
+        cls.create_pool_kwargs = {'listener_id': cls.listener_id,
+                                  'protocol': "HTTP",
+                                  'lb_algorithm': "ROUND_ROBIN"}
+        cls.pool = (
+            cls._create_pool(**cls.create_pool_kwargs))
+        cls.pool_id = cls.pool['id']
+
+        # Create basic args for policy creation
+        cls.create_l7policy_kwargs = {'listener_id': cls.listener_id,
+                                      'admin_state_up': "true"}
+
+        # Get a client to emulate the agent's behavior.
+        cls.client = cls.plugin_rpc.get_client()
+        cls.context = cls.plugin_rpc.get_context()
+
+    @classmethod
+    def resource_cleanup(cls):
+        super(L7PolicyTestJSON, cls).resource_cleanup()
+
+    @test.attr(type='smoke')
+    def test_create_l7_reject_policy(self):
+        """Test the creationg of a L7 reject policy."""
+        create_l7policy_kwargs = self.create_l7policy_kwargs
```

Comment:
You may want to import copy and do copy.deepcopy of that dict to make a sufficient copy to amend. Otherwise, these names are referencing the same object (dict).